HW-12/main.py

```diff
@@ -0,0 +1,163 @@
+
+    command_splitted = command.split(" ", 1)
+    if command_splitted[0] == 'delete' and len(command_splitted) > 1:
+        splitted_value = command.split(" ", 1)[1]
+        my_list = phonebook.keys()
+        if splitted_value in my_list:
+            del phonebook[splitted_value]
+            with open('phonebook.json', 'w') as file:
+                json.dump(phonebook, file)
+            print(f'{splitted_value} successfully deleted')
+            print(f'Now you have {len(phonebook)} contacts')
+            print(phonebook)
+            command = ''
+            continue
+        else:
+            print(f'There is no contact {splitted_value}')
+            command = ''
+            continue
+    elif command_splitted[0] == 'delete':
+        print("All data has been deleted")
+        phonebook.clear()
+        with open('phonebook.json', 'w') as file:
+            json.dump(phonebook, file)
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    command_splitted = command.split(" ", 1)
+    if command_splitted[0] == 'show' and len(command_splitted) > 1:
+        splitted_value = command.split(" ", 1)[1]
+        my_list = phonebook.keys()
+        if splitted_value in my_list:
+            print(f'Here is full info about user {splitted_value}: {phonebook.get(splitted_value)}')
+            command = ''
+            continue
+        else:
+            print(f'There is no contact {splitted_value}')
+            command = ''
+            continue
+    elif command_splitted[0] == 'show':
+        answer = input('Please enter name to show: ')
+        if answer in phonebook.keys():
+            print(f'Here is full info about user {answer}: {phonebook.get(answer)}')
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'list':
+        if len(phonebook) == 0:
+            print('Phonebook is empty')
+        else:
+            for key in phonebook:
+                print(key)
+                command = ''
+            continue
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'exit':
+        break
+
+
+# 2. Написати декоратор, який буде записувати в файл назву функції, яку він декорує, і писати час її виклику.
+
+def decorator(func):
+    def wrapper(*args, **kwargs):
+        print(f'Function name: {func.__name__}')
+        print(f'Function was called at: {time.strftime("%d/%m/%Y, %H:%M:%S")}')
+        return func
+
+    return wrapper
+
+
+@decorator
+def calc():
+    print("d")
+
+
+calc()
+
+
+# 3. В попередньо написаний кастомний Exception додати запис помилки і час її виникнення у файл.
+
+class MyException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        with open('exception.txt', 'a') as file:
```

Comment:
Change the modifier to `a+` to avoid an error if the file does not exist

Reply:
Changed. I will study the topic again

Reply:
The same mistake for `a`: this modifier allows to work with non-existing files. The difference between `a` and `a+` is that  `a+` allows to read from file but not only write to it